include/ebpf_windows.h

```diff
@@ -70,9 +70,30 @@ typedef enum _ebpf_helper_function
 
 #define EBPF_MAX_GENERAL_HELPER_FUNCTION 0xFFFF
 
-#define EBPF_ATTACH_CLIENT_DATA_CURRENT_VERSION 1
-#define EBPF_PROGRAM_INFORMATION_CLIENT_DATA_CURRENT_VERSION 1
+// Warning: Version 1 of the eBPF extension data structures will be deprecated in a future release.
+// All extension data structures should be updated to version 2.
+// Leaving version 1 for backward compatibility for now.
```

Comment:
Conditionally supplying either version 1 or version 2 based on user mode behavior (i.e., whether a load flag was set) seems needlessly dangerous: the repurposing of an existing field in version 2 implies that anyone using version 1 now, without very careful version checks, will likely have subtle bugs leading to poorly defined behavior.

I suggest making a breaking change now and removing version 1.

Reply:
Ah, I've read the eBPF extension documentation more carefully, and it is very clear that the version number change implies a breaking change to the struct. I'm OK with this as-is.

Reply:
I am fine dropping 1 completely, was just hoping to do it over N releases so extensions can be updated.

Reply:
Allow me to partially recant: I've already found at least one extension incorrectly validating headers provided by eBPF. 

That doesn't shift the responsibility onto eBPF, but again, I think it is better to force extensions onto version 2, rather than leave the struct version conditional on user mode behavior.

Reply:
New capabilities-based approach should fix this.